Make install --quiet really quiet

njsmith opened the pull request:
Commit 5bb989993 added some code for hiding/showing the cursor when
running on the terminal, but accidentally made it so that it always
printed the invisible control codes, even when not on a tty or when
--quiet was specified. This turns out to be surprisingly bad (e.g. it
breaks the official docker python builds). So, let's not do that.

Fixes gh-3418

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/pypa/pip/3444)
<!-- Reviewable:end -->

njsmith commented:
(I guess this probably needs a backport for 8.0.3, given the docker issue.)

xavfernandez commented on pip/utils/ui.py:
```diff
@@ -215,10 +215,16 @@ def update(self):
 
 @contextlib.contextmanager
 def hidden_cursor(file):
+    magic_bytes = (HIDE_CURSOR, SHOW_CURSOR)
```
Are these used ?

njsmith replied:
Doh, thought I deleted that :-) fixed

xavfernandez commented:
Some pep8 issues apparently and the `magic_bytes` question but otherwise seems fine :+1: 
A changelog would also be nice :)

njsmith commented:
Addressed all comments, I believe.

I just put the changelog in the 8.1.0 section -- don't know if you want to do a 8.0.3 or not, but I guess I'll let you figure that out :-)

xavfernandez commented:
Thanks :)
